### DISCUSSION TITLE
Pinia with Nuxt3 causing large bundle size when deployed to AWS Lambda

### DISCUSSION THREAD

**Discussion by mferraco** [vuejs/pinia#1764] [Help and Questions]
## Context

I am building a Vue3/Nuxt3 project using Pinia and set up the project as defined [here](https://pinia.vuejs.org/ssr/nuxt.html).

I am deploying my project to aws-lambda and I'm using [vite bundler](https://vitejs.dev/) that comes packaged with Nuxt to create the project bundle that is uploaded to AWS.

## Problem

When this project is deployed **with** `@pinia/nuxt` included as a dependency, the deployed project size in aws lambda is 33mb.
When this project is deployed **without** `@pinia/nuxt` included as a dependency, the deployed project in aws lambda is 9.3 mb.

This is an issue because AWS Lambda has a [deployment package max size of 50mb zipped](https://docs.aws.amazon.com/lambda/latest/dg/gettingstarted-limits.html#function-configuration-deployment-and-execution).

After digging into the files included in the deployed bundle, I realized that the `@pinia/nuxt` package is including `typescript` as a dependency in the deployed package on aws lambda. The typescript package alone results in >70% of the 33mb package size when `@pinia/nuxt` is included in the project. It disappears when `@pinia/nuxt` is removed from the project. This package is a dev dependency and shouldn't be in the deployment package.

## Question

**Why is `typescript` being included as a dependency when deploying my project with `@pinia/nuxt`?**

**Comment by posva** [comment]
TypeScript is probably brought in by `@nuxt/kit` somewhere down in its dependency tree.
When you deploy your application, are you packing your node_modules? I think you only need to deploy the `.output` folder. Although, there is a preset for AWS: https://v3.nuxtjs.org/getting-started/deployment/#deployment
